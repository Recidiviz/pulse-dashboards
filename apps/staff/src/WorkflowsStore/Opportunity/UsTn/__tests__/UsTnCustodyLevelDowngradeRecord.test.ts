// Recidiviz - a data platform for criminal justice reform
// Copyright (C) 2022 Recidiviz, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
// =============================================================================

import {
  UsTnCustodyLevelDowngradeReferralRecordRaw,
  usTnCustodyLevelDowngradeSchema,
} from "../UsTnCustodyLevelDowngradeOpportunity/UsTnCustodyLevelDowngradeReferralRecord";

const usTnCustodyLevelDowngradeRecordRaw: UsTnCustodyLevelDowngradeReferralRecordRaw =
  {
    stateCode: "US_XX",
    externalId: "abc123",
    eligibleCriteria: {
      custodyLevelHigherThanRecommended: {
        recommendedCustodyLevel: "HIGH",
        custodyLevel: "MAXIMUM",
      },
      custodyLevelIsNotMax: null,
      usTnLatestCafAssessmentNotOverride: {
        overrideReason: "Some reason",
      },
      usTnIneligibleForAnnualReclassification: {
        ineligibleCriteria: ["Some reason"],
      },
    },
    ineligibleCriteria: {},
    formInformation: {
      activeRecommendations: [],
      classificationType: "SPECIAL",
      hasIncompatibles: false,
      incompatibleArray: [],
      statusAtHearingSeg: "GEN",
      currentOffenses: ["ROBBERY-ARMED WITH DEADLY WEAPON"],
      lastCafDate: "2022-08-22",
      lastCafTotal: "8",
      q1Score: 1,
      q2Score: 1,
      q3Score: 1,
      q4Score: 1,
      q5Score: 1,
      q6Score: 1,
      q7Score: 1,
      q8Score: 1,
      q9Score: 1,
      q6Notes: [{ eventDate: "2022-08-22", noteBody: "Some note" }],
      q7Notes: [{ eventDate: "2022-08-22", noteBody: "Some note" }],
      q8Notes: [
        {
          detainerReceivedDate: "2022-08-22",
          detainerFelonyFlag: "X",
          detainerMisdemeanorFlag: "X",
        },
      ],
    },
  };

test("record is properly parsed for opportunity", () => {
  expect(
    usTnCustodyLevelDowngradeSchema.parse(usTnCustodyLevelDowngradeRecordRaw),
  ).toMatchSnapshot();
});
