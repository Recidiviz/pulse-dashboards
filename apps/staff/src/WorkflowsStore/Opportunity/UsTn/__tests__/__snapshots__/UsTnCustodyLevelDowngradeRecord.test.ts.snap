// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`record is properly parsed for opportunity 1`] = `
Object {
  "caseNotes": Object {
    "ASSAULTIVE DISCIPLINARIES": Array [],
  },
  "eligibleCriteria": Object {
    "custodyLevelHigherThanRecommended": Object {
      "custodyLevel": "MAXIMUM",
      "recommendedCustodyLevel": "HIGH",
    },
    "custodyLevelIsNotMax": null,
    "usTnIneligibleForAnnualReclassification": Object {
      "ineligibleCriteria": Array [
        "Some reason",
      ],
    },
    "usTnLatestCafAssessmentNotOverride": Object {
      "overrideReason": "Some reason",
    },
  },
  "externalId": "abc123",
  "formInformation": Object {
    "activeRecommendations": Array [],
    "classificationType": "SPECIAL",
    "currentOffenses": Array [
      "ROBBERY-ARMED WITH DEADLY WEAPON",
    ],
    "hasIncompatibles": false,
    "incompatibleArray": Array [],
    "lastCafDate": 2022-08-22T00:00:00.000Z,
    "lastCafTotal": "8",
    "q1Score": 1,
    "q2Score": 1,
    "q3Score": 1,
    "q4Score": 1,
    "q5Score": 1,
    "q6Notes": Array [
      Object {
        "eventDate": 2022-08-22T00:00:00.000Z,
        "noteBody": "Some note",
      },
    ],
    "q6Score": 1,
    "q7Notes": Array [
      Object {
        "eventDate": 2022-08-22T00:00:00.000Z,
        "noteBody": "Some note",
      },
    ],
    "q7Score": 1,
    "q8Notes": Array [
      Object {
        "detainerFelonyFlag": true,
        "detainerMisdemeanorFlag": true,
        "detainerReceivedDate": 2022-08-22T00:00:00.000Z,
      },
    ],
    "q8Score": 1,
    "q9Score": 1,
    "statusAtHearingSeg": "GEN",
  },
  "ineligibleCriteria": Object {},
  "stateCode": "US_XX",
}
`;
